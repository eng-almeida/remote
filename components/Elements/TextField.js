import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';

import Field from './Field';
import { FieldText } from './Typography';

const rotate = keyframes`
  from {
    width: 0;
  }

  to {
    width: 100%;
  }
`;

const Border = styled.div`
	position: absolute;
	bottom: 0;
	width: 0;
	border-bottom: 1px solid ${({ theme }) => theme.colors.purple[200]};
`;

const TextFieldContainer = styled.div`
	position: relative;
`;

const TextInput = styled.input`
	${FieldText}
 	font-family: inherit;
	color: ${({ theme }) => theme.colors.grays[300]};
	outline: none;
	background: transparent;
	border: none;
	border-bottom: 1px solid ${({ theme }) => theme.colors.grays[400]};
	padding-bottom: ${({ theme }) => theme.space[2]};
	width: 100%;

	&::placeholder,
	&::-webkit-input-placeholder,
	&:-ms-input-placeholder {
		color: ${({ theme }) => theme.colors.grays[200]};
	}

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&[type=number] {
		-moz-appearance: textfield;
	}

	&:focus + ${Border} {
		animation: ${rotate} 0.2s linear;
		animation-fill-mode: forwards;
	}
`;

const TextField = ({ label, helperText, ...inputProps }) => (
	<Field label={label} helperText={helperText} id={inputProps.id}>
		<TextFieldContainer>
			<TextInput {...inputProps} />
			<Border />
		</TextFieldContainer>
	</Field>
);

TextField.propTypes = {
	label: PropTypes.string,
	helperText: PropTypes.string,
};

export default TextField;