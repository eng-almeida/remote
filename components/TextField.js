import styled from 'styled-components';

import Field from './Field';

const TextInput = styled.input`
 	font-family: inherit;
	font-size: 20px;
	line-height: 24px;
	color: ${({ theme }) => theme.colors.grays[300]};
	outline: none;
	background: transparent;
	border: none;
	border-bottom: 1px solid ${({ theme }) => theme.colors.grays[400]};
	padding-bottom: ${({ theme }) => theme.space[2]};
	width: 100%;

	&::placeholder,
	&::-webkit-input-placeholder,
	&:-ms-input-placeholder {
		color: ${({ theme }) => theme.colors.grays[200]};
	}

	&:focus {
		border-bottom: 1.5px solid ${({ theme }) => theme.colors.purple[200]};
	}
`;

const TextField = ({ label, helperText, ...inputProps }) => {
  return (
    <Field label={label} helperText={helperText} id={inputProps.id}>
      <TextInput {...inputProps} />
    </Field>
  )
}

export default TextField;