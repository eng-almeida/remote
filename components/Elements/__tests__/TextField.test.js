import TextField from '@/components/Elements/TextField';

describe('TextField component', () => {
	it('should match snapshot', () => {
		const { container } = renderWithProviders(
			<TextField
				label="Test"
				helperText="Test helper"
				id="test"
				name="test"
				placeholder="Test"	
			/>
		);
		expect(container).toMatchSnapshot();
	});
})