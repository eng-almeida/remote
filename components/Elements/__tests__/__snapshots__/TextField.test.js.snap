// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component should match snapshot 1`] = `
.c0 {
  font-size: 13px;
  line-height: 16px;
  display: block;
  color: #778CA3;
  margin-bottom: 4px;
  -webkit-transition: color 0.2s;
  transition: color 0.2s;
}

.c5 {
  font-size: 13px;
  line-height: 16px;
  color: #00234B;
  opacity: 0.5;
  margin-top: 8px;
}

.c4 {
  position: absolute;
  bottom: 0;
  width: 0;
  border-bottom: 1px solid #624DE3;
}

.c1 {
  position: relative;
}

.c2 {
  font-size: 20px;
  line-height: 24px;
  font-family: inherit;
  color: #00234B;
  outline: none;
  background: transparent;
  border: none;
  border-bottom: 1px solid #CED5DB;
  padding-bottom: 8px;
  width: 100%;
}

.c2::-webkit-input-placeholder,
.c2::-webkit-input-placeholder,
.c2:-ms-input-placeholder {
  color: #525F7F;
}

.c2::-moz-placeholder,
.c2::-webkit-input-placeholder,
.c2:-ms-input-placeholder {
  color: #525F7F;
}

.c2:-ms-input-placeholder,
.c2::-webkit-input-placeholder,
.c2:-ms-input-placeholder {
  color: #525F7F;
}

.c2::placeholder,
.c2::-webkit-input-placeholder,
.c2:-ms-input-placeholder {
  color: #525F7F;
}

.c2::-webkit-outer-spin-button,
.c2::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c2[type=number] {
  -moz-appearance: textfield;
}

.c2:focus + .c3 {
  -webkit-animation: cvpVum 0.2s linear;
  animation: cvpVum 0.2s linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

<div>
  <div>
    <label
      class="c0"
      for="test"
    >
      Test
    </label>
    <div
      class="c1"
    >
      <input
        class="c2"
        id="test"
        name="test"
        placeholder="Test"
      />
      <div
        class="c3 c4"
      />
    </div>
    <div
      class="c5"
    >
      Test helper
    </div>
  </div>
</div>
`;
